tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TesTaskServiceImpl.java

```diff
@@ -57,8 +68,38 @@ public TesCreateTaskResponse submitTesTask(TesTask body) {
     }
 
     @Override
-    public TesListTasksResponse listTesTask() {
-        return new TesListTasksResponse();
+    public TesListTasksResponse listTesTask(String namePrefix, Long pageSize, String pageToken, TaskView view) {
+        TesListTasksResponse tesListTasksResponse = new TesListTasksResponse();
+        List<PipelineRun> pipelineRunList;
+        if (StringUtils.isNotEmpty(namePrefix)) {
+            pipelineRunList = searchRunsWithNamePrefix(namePrefix, pageSize, pageToken);
+        } else {
+            pipelineRunList = filterRunsWithOutNamePrefix(pageSize, pageToken);
+        }
+        tesListTasksResponse.setTasks(pipelineRunList.stream().map(pipelineRun ->
+                taskMapper.mapToTesTask(pipelineRun, Optional.ofNullable(view).orElse(DEFAULT_TASK_VIEW)))
+                .collect(Collectors.toList()));
+        return tesListTasksResponse;
```

Comment:
can we return here somethink like `new TesListTasksResponse(pipelineRunList)`? In order to get rid of tesListTasksResponse local var

Reply:
done

Reply:
Added @NonNull annotation for first (List<TesTask> tasks) argument in TesListTasksResponse class, to be able to use constructor with only one(this) argument